hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -667,6 +676,11 @@ BlobProperty getBlobProperty(Path blobPath,
     blobProperty.setCopyStatus(opResult.getResponseHeader(X_MS_COPY_STATUS));
     blobProperty.setContentLength(
         Long.parseLong(opResult.getResponseHeader(CONTENT_LENGTH)));
+
+    if (blobPath.isRoot()) {
+      blobProperty.setContentLength(0L);
+      blobProperty.setIsDirectory(true);
```

Comment:
Because getContainerProperty does not have x-ms-metada-hdi-isfolder in response. This is only for blobs.
Container is not a directory in Azure terms but is a directory in fileSystem terms.